File: botan.js
var request = require("request");

var url_template = 'http://olmer.haze.yandex.net/botan?token=#token#&uid=#uid#&name=#name#';

module.exports = function(apikey) {
    token = apikey;
    return {
        track: function(message, name) {
            uid = message.from.id;
            name = name ? name : 'Message';
            url = url_template.replace('#token#', token).replace('#uid#', uid).replace('#name#', name);
            //console.log(url);
            request({
                    url: url,
                    method: "POST",
                    json: message
                }, function (error, response, body) {
                    if (!error && response.statusCode === 200) {
                        //console.log(body);
                    }
                    else {
                        console.log("error: " + error);
                        console.log("response.statusCode: " + response.statusCode);
                        console.log("response.statusText: " + response.statusText);
                    }
                }
            );
        }
    };
};

File: botan.rb
#usage example:
#
#require_relative 'botan'
#
#puts Botan.track(1111, 1, {'text' => 2}, 'Search')
#
module Botan
  require 'net/http'
  require 'net/https'
  require 'json'

  URI_TEMPLATE = 'http://retention.yurich.serp.yandex.ru/botan?token=%{token}&uid=%{uid}&name=%{name}';

  def self.track(token, uid, message, name = 'Message')
    begin
      uri = URI(URI_TEMPLATE % {token: token, uid: uid, name: name})
      puts uri
      http = Net::HTTP.new(uri.host, uri.port)
      body = JSON.dump(message)

      req =  Net::HTTP::Post.new(uri)
      req.add_field "Content-type", "application/json"
      req.body = body
      # Fetch Request
      res = http.request(req)
      return JSON.parse(res.body)
    rescue StandardError => e
      puts "HTTP Request failed (#{e.message})"
    end
  end
end
